README/ReleaseNotes/v612/index.md

```diff
@@ -112,40 +112,46 @@ large TClonesArray where each element contains another small vector container.
+
+#### New features
+  - Add `Alias`, a facility to specify an alternative name for a given column: `auto histo = mytdf.Alias("myAlias", "myColumn").Histo1D("myAlias");`
+  - Add `Cache`, a facility to cache `TDataFrame`s in memory. All or some columns can be cached. Two versions of the method are proposed: one which allows to explicitly list the types of the columns and another one allowing to let the system infer them (the same mechanism of the `Snapshot` method). Only columns containing instances of classes which have a copy constructor can be cached.
+  - Add `DefineSlot`, a `Define` transformation that is aware of the multi-threading slot where the workload is executed
+  - Add `DefineSlotEntry`, a `Define` transformation that is aware of the multi-threading slot and of the current entry number
+  - Add `GetColumnsNames`: users can now get the names of the available columns coming from trees, data sources or `Define`d columns
   - Add `OnPartialResult` and `OnPartialResultSlot`: users can now register one or more functions to be executed on partial results of TDF actions during the event loop.
     This mechanism is meant to be used to inspect partial results of the analysis or print useful debug information.
     For example, both in single- and multi-thread event loops, one can draw a result histogram and update the canvas every 100 entries like this:
-    ```
+    ```c++
     auto h = tdf.Histo1D("x");
     TCanvas c("c","x hist");
     h.OnPartialResult(100, [&c](TH1D &h_) { c.cd(); h_.Draw(); c.Update(); });
     ```
     See the tutorials for more examples.
-  - Added `DefineSlotEntry`, a `Define` transformation that is aware of the multi-threading slot and of the current entry number
-  - Users can now access multi-threading slot and entry number as pre-defined columns "tdfslot_" and "tdfentry_"
-  - Added a `Sum` action that sums all values of a column for the processed entries
-  - Added support for friend trees in all cases
+  - Add `Sum`, an action that sums all values of a column for the processed entries
+  - The new TDataSource interface allows developers to pipe any kind of columnar data format into TDataFrame. Two example data sources have been provided: the TRootDS and the TTrivialDS. The former allows to read via the novel data source mechanism ROOT data, while the latter is a simple generator, created for testing and didactic purposes. It is therefore now possible to interface *any* kind of dataset/data format to ROOT as long as an adaptor which implements the pure virtual methods of the TDataSource interface can be written in C++.
+  - TDF can now read CSV files through a specialized TDataSource. Just create the TDF with `MakeCsvDataFrame("f.csv", false, ',')`
```

Comment:
Out of the three parameters of `MakeCsvDataFrame("f.csv", false, ',')`, only the first one is actually needed. What about this: "Just create the TDF with `MakeCsvDataFrame("f.csv")`. The data types of the CSV columns are automatically inferred. You can also specify if want to use a different delimiter or if your file does not have headers."

Reply:
Hi, I had checked [here](https://root.cern/doc/master/namespaceROOT_1_1Experimental_1_1TDF.html#a85b75a97f36b32a9dfd7627e795e95ca) where it seems all three parameters are required

Reply:
The documentation of the parameters mentions their default values and they are also present in the corresponding header, but it is true that those default values don't appear in the function signature in the docs. How does Doxygen pick them?

Reply:
Ah this is the usual problem with putting doxygen in the cxx files. @Axel-Naumann what's the rationale? Afaik the common practice is putting the doxygen docs in the header file (where it can also pick up the default values of function parameters)